chart/templates/_helpers.tpl

```diff
@@ -77,8 +77,20 @@ Usage:
 {{ include "base_pull_secrets" . }}
 */}}
 {{- define "base_pull_secrets" -}}
-    {{- if .Values.base.imagePullSecrets.enabled }}
-    {{- include "render" (dict "value" .Values.base.imagePullSecrets.secrets "context" $) | nindent 8 }}
+    {{- if (not (empty .Values.image.pullSecrets)) }}
+        {{- range .Values.image.pullSecrets | uniq -}}
+            {{ nindent 8 "- name:" }} {{ . }}
+        {{- end }}
+    {{- else -}}
+        {{- if .Values.base.imagePullSecrets }}
+            {{- if .Values.base.imagePullSecrets.enabled }}
+                {{- if (empty .Values.base.imagePullSecrets.secrets) }}
+                    {{ nindent 8 "- name: login" }}
```

Comment:
Is this really necessary? If neither the new nor the old configuration keys are being used, should we assume configuration for a user? This feels like it is a good candidate for a failure case.

Reply:
I was thinking of the case where the user is specifying --set base.imagePullSecrets=true but relying of the default name of "login", and having defined a secret called login in the cluster. In which case the name would no longer be defined in the updated values.yaml file.
I couldn't say how likely this scenario would be for users, but is possible AFAICT.

Reply:
Okay, yeah that case makes sense to me. Good idea to leave this in I guess.